vendor/api/v1/stripe/views.py

```diff
@@ -0,0 +1,194 @@
+import json
+import logging
+import stripe
+
+from django.conf import settings
+from django.core.exceptions import ObjectDoesNotExist
+from django.http.response import HttpResponse
+from django.db.models import TextChoices
+from django.utils.translation import gettext as _
+from django.views import View
+from django.views.decorators.csrf import csrf_exempt
+
+from vendor.models.choice import InvoiceStatus
+from vendor.integrations import StripeIntegration
+from vendor.utils import get_site_from_request
+from vendor.models import CustomerProfile, Subscription, Invoice
+from vendor.processors import StripeProcessor
+
+
+logger = logging.getLogger(__name__)
+
+
+class StripeEvents(TextChoices):
+    INVOICE_PAID = 'invoice.paid', _('Invoice Paid')
+    INVOICE_PAYMENT_FAILED = 'invoice.payment_failed', _('Invoice Payment Failed')
+
+class StripeBaseAPI(View):
+
+    def __init__(self, **kwargs):
+        self.stripe_event = None  # Variable used to store the webhooks event data.
+
+        super().__init__(**kwargs)
+
+    @csrf_exempt
+    def dispatch(self, *args, **kwargs):
+        return super().dispatch(*args, **kwargs)
+
+    def is_valid_post(self, site):
+        try:
+            credentials = StripeIntegration(site)
+
+            if credentials.instance and credentials.instance.private_key:
+                self.event = stripe.Event.construct_from(json.loads(self.request.body), credentials.instance.private_key)
+            elif settings.STRIPE_PUBLIC_KEY:
+                self.event = stripe.Event.construct_from(json.loads(self.request.body), settings.STRIPE_PUBLIC_KEY)
+            else:
+                return False
+            
+        except ValueError as exce:
+            return False
+        except Exception as exce:
+            return False
+
+        return True
+    
+    def is_incoming_event_correct(self, event, desired_event):
+        if event.type != desired_event:
+            return False
+
+        # This check is recuired to make sure that the event is related to a subscription.
+        if self.event.data.object.billing_reason != 'subscription_cycle':
+            return False
+
+        return True
+
+
+class StripeSubscriptionInvoicePaid(StripeBaseAPI):
+
+    def post(self, request, *args, **kwargs):
+        site = get_site_from_request(self.request)
+
+        if not self.is_valid_post(site):
+            return HttpResponse(status=400)
+
+        if not self.is_incoming_event_correct(self.event, StripeEvents.INVOICE_PAID):
+            return HttpResponse(status=400)
+
+        # TODO nice to move this assignments inside a generic function in the StripeBaseAPI class
+        stripe_subscription_id = self.event.data.object.subscription
```

Comment:
not a big deal but if you wanted to assign a self.event.data.object to a variable and use that to get the following data, it might make this a bit more readable. eg: 
```
data_object = self.event.data.object
stripe_subscription_id = data_object.subscription
stripe_customer_id = data_object.customer
...
```